### ISSUE TITLE
Missing poligons on .obj files.

### ISSUE THREAD

**Issue by Appsepsi** [armory3d/armorpaint#411]
This happens when I export .obj files from Cinema 4D and use them in ArmorPaint. There are always missing poligons and uv maps are also screwed up.

Armorpaint version : ArmorPaint v0.8 (2020-05-10)
Os : Windows - Direct3D11
Gpu : Nvidia GeForce GTX 1650
Cpu : Intel Core i5-9300H

**Comment by luboslenco** [comment]
Please always throw in a sample .obj file for issues like these, so I can reproduce and fix it. Thanks!

**Comment by Appsepsi** [comment]
https://drive.google.com/file/d/1weoenLIa4g9RjLDZF4H7A2QfRMfy0tOF/view?usp=sharing
This is an example object made in Cinema 4D r19 where the bug is visible, but i also got glitches on a simple plane from Cinema 4D.
![1](https://user-images.githubusercontent.com/52209860/82680937-4e66db80-9c4d-11ea-9616-8c80833ff4d4.PNG)

**Comment by luboslenco** [comment]
Thanks for the file! The missing faces appear to be caused by reversed normals & backface culling. Quick test in Blender with backface culling enabled:

![cull copy](https://user-images.githubusercontent.com/3685704/83394269-9679c100-a3f8-11ea-9f89-77285d8ee700.jpg)

Reversing normals for the affected faces should resolve the issue.

There is a way to disable backface culling in ArmorPaint at `Menu bar - Viewport - Cull Backfaces`, however it may cause issues with lighting due to normals pointing in the opposite direction (and gives worse performance).

Related:
- https://www.schoolofmotion.com/blog/missing-objects-cinema-4d - 4. REVERSED NORMALS & BACKFACE CULLING
- https://en.wikipedia.org/wiki/Back-face_culling

**Comment by Appsepsi** [comment]
Yes, you are right. I didn´t notice inverted normals because in cinema 4D there is no backface culling.
Thanks for your reply.
But I still have one issue with uv maps. When I paint somewhere, it also paints somewhere else. On screenshot (below) there is a plane with a cone, I painted one edge of the plane with white color and it painted every edge of the plane. I tryed uv unwarping it with built in plugin in Armorpaint but it didn´t help. Is there any way to fix this?

![1](https://user-images.githubusercontent.com/52209860/83405505-ea8ea080-a40c-11ea-81ce-aa77135a5a4a.PNG)

**Comment by luboslenco** [comment]
> When I paint somewhere, it also paints somewhere else.

My guess without seeing the file is that there are overlapping UVs - multiple faces share the same position on the uv map. It can be resolved by unwrapping the mesh so that each face occupies unique space on the uv map, or using multiple layers to paint objects with multiple uv sets like shown [here](https://www.youtube.com/watch?v=ocIPd6sgQBs&list=PLsoJIZPOq9_r2aciBXDB0VBo3Z0lTI2En&index=22).

Not sure about the unwrap plugin producing incorrect uv map, please feel welcome to open a new issue on that.

**Comment by Appsepsi** [comment]
Yes, thanks. As you told it´s because of overlapping UVs. But the plugin seems like it´s not working at all. I tryed unwraping the mesh and I didn´t see any changes at first and the painting made it even worse. There is the object if you want to see it (link below), but it can be also seen on the chair that I sent and the uv unwrap plugin didn´t fix the issue. I think it´s better to test it on the chair, because the last shown object isn´t very good and it´s not finished. I exported it just to show the problem. When I unwraped the chair I got the same problem, but on different spots. Somewhere I couldn´t even paint at all. It looked like there were faces without uv map.

http://leteckaposta.cz/494987670
